lib/fibonacci.rb

```diff
@@ -1,8 +1,22 @@
 # Improved Fibonacci
 
-# Time Complexity - ?
-# Space Complexity - ?  (should be O(n))
-# Hint, you may want a recursive helper method
+# Time Complexity - O(n)
+# Space Complexity - O(1)? <-- I replace the O(n) array into just keeping track of the last two 
+# Hint, you may want a recursive helper method 
 def fibonacci(n)
```

Comment:
You still have the system stack for space complexity, but otherwise this is well done.